apps/legal_archive/commands.py

```diff
@@ -332,12 +332,12 @@ class ImportLegalPublishQueueCommand(superdesk.Command):
     def run(self):
         logger.info('Import to Legal Publish Queue')
         lock_name = get_lock_id('legal_archive', 'import_legal_publish_queue')
-        if not lock(lock_name, '', expire=600):
+        if not lock(lock_name, 'superdesk', expire=600):
             return
         try:
             LegalArchiveImport().import_legal_publish_queue()
         finally:
-            unlock(lock_name, '')
+            unlock(lock_name, 'superdesk')
```

Comment:
@petrjasek That's true. We were thinking to set the lock expiry equal to the soft time limit to handle this problem. I think for the en-queue and transmit tasks this is the case (we can make it for the legal_archive_import as well). I'll go through other lock cases as well.